apps/src/code-studio/pd/workshop_dashboard/reports/foorm/results_loader.jsx

```diff
@@ -0,0 +1,98 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import $ from 'jquery';
+import Spinner from '../../../components/spinner';
+import Results from './results';
+import color from '@cdo/apps/util/color';
+
+const styles = {
+  errorContainer: {
+    marginTop: 15,
+    marginLeft: 15
+  },
+  errorDetailsBox: {
+    backgroundColor: color.lightest_gray,
+    padding: 20,
+    maxWidth: 550,
+    marginTop: 20
+  }
+};
+
+export class ResultsLoader extends React.Component {
+  static propTypes = {
+    params: PropTypes.shape({
+      workshopId: PropTypes.string.isRequired
+    })
+  };
+
+  state = {loading: true, errors: null};
+
+  componentDidMount() {
+    this.load();
+  }
+
+  load() {
+    const url = `/api/v1/pd/workshops/${
+      this.props.params['workshopId']
+    }/foorm/generic_survey_report`;
+
+    this.loadRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    })
+      .done(data => {
+        this.setState({
+          loading: false,
+          questions: data['questions'],
+          thisWorkshop: data['this_workshop'],
+          sessions: Object.keys(data['this_workshop']),
```

Comment:
Is `data['this_workshop']` always a hash of sessions?

Reply:
I will rename this prop to make it more accurate, because the answer is...kind of. It's usually a hash of sessions, but may also include pre/post surveys or just one set of survey results if the workshop is only a 1-day workshop. Does workshopTabs make more sense? It will end up being the names of the non-rollup tabs:
![Screen Shot 2020-03-17 at 4 00 24 PM](https://user-images.githubusercontent.com/33666587/76909445-7bbb9580-6868-11ea-89d9-88a62244ed4d.png)

Reply:
I think at this point in the pipeline we should be UI agnostic, and name it for what kind of data it represents.  Not that I know what that should be?  Collection?  Segments?  Result bucket?

Reply:
the backend itself won't have a name for this, since it's just the keys for this_workshop. On this side the only place workshopTabs would be used is on the component that creates the tabs

Reply:
Oh, for some reason I thought this was in .rb.  Yes, that's a fair argument for `workshopTabs` then!